dialogue-target/src/main/java/com/palantir/dialogue/Request.java

```diff
@@ -187,6 +188,7 @@ public Request.Builder putHeaderParams(String key, String... values) {
         }
 
         public Request.Builder putHeaderParams(String key, String value) {
+            Preconditions.checkArgumentNotNull(key, "Header name must not be null");
```

Comment:
These should probably be `checkNotNull`:
- It's what we use in our other builders and constructors
- If one of these values is null it's very likely an application bug and not due to user input